### ISSUE TITLE
Example of how to write binary download to disk ?

### ISSUE THREAD

**Issue by Rockvole** [flutterchina/dio#15]
Hi,
I am trying to download a zipfile and store the resulting zip file to disk.

    Options options= new Options(
        responseType: ResponseType.STREAM
    );
    Dio dio = new Dio(options);
    Response response=await dio.get(url);
    File file = new File(outputDirectory+ filename);
    await file.writeAsBytes(response.data);

writeAsBytes doesnt seem to return.

Do you have an example of how to do this ?

**Comment by wendux** [comment]
Why not use `download` API  directly ?

**Comment by Rockvole** [comment]
Ok, thanks for the tip.
I tried :

    Options options= new Options(
        responseType: ResponseType.STREAM
    );
    Dio dio = new Dio(options);
    try {
      Response response = await dio.download(
          url, outputDirectory + filename, flush: true);
      dio.interceptor.response.onError = (DioError e) {
        print("error=" + e.message);
      };
    } on DioError catch (e) {
      print("error "+e.message);
    }

It does download the file, but is never returning from the download method.

**Comment by wendux** [comment]
could  you print the response ?


 📱[From gitme Android](http://flutterchina.club/app/gm.html)

**Comment by Rockvole** [comment]
I don't believe there was any error, although that was a month ago. I have just used the standard dart mechanism instead.
